source_gen/lib/src/output_helpers.dart

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:async/async.dart';
+
+/// Runs [run] and converts [Future], [Iterable], and [Stream] implementations
+/// containing [String] to a single [Stream] while ensuring all thrown
+/// exceptions are forwarded through the return value.
+Stream<String> normalizeGeneratorOutput(Object run()) => StreamCompleter
+    .fromFuture(new Future.sync(run).then(_normalizeGeneratorOutput));
+
+Stream<String> _normalizeGeneratorOutput(Object value) {
```

Comment:
nit: `dynamic` I think it techinically the right thing, you don't allow any type of Object.